### PR TITLE
[WFLY-14281] - Integrate MP Fault Tolerance 3.0 into EAP

### PR DESCRIPTION
**Pull request by jasondlee**
Integrate MP Fault Tolerance 3.0 into Wildfly

JIRA: https://issues.redhat.com/browse/WFLY-14281
Analysis Doc PR: https://github.com/wildfly/wildfly-proposals/pull/361

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -312,12 +312,13 @@
         <version.io.rest-assured>3.0.6</version.io.rest-assured>
         <version.io.smallrye.smallrye-common>1.5.0</version.io.smallrye.smallrye-common>
         <version.io.smallrye.smallrye-config>2.0.2</version.io.smallrye.smallrye-config>
-        <version.io.smallrye.smallrye-fault-tolerance>4.3.2</version.io.smallrye.smallrye-fault-tolerance>
         <version.io.smallrye.smallrye-health>3.0.0-RC2</version.io.smallrye.smallrye-health>
+        <version.io.smallrye.smallrye-fault-tolerance>5.0.0-RC3</version.io.smallrye.smallrye-fault-tolerance>
```
Comment (kabir) [inline pom.xml:316]: Keep in alphabetical order

**Review by kabir — CHANGES_REQUESTED** [review]
The new dependencies need to go into `microprofile/galleon-common/src/license/microprofile-feature-pack-licenses.xml`

File: pom.xml (hunk 1 above)
Comment (rhusar) [inline pom.xml:316]: The 5.0 final is out now that we have been waiting for.

**Review by rhusar — CHANGES_REQUESTED** [review]

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/fault-tolerance/main/module.xml [hunk 2]
```diff
@@ -36,6 +36,8 @@
     <dependencies>
         <module name="io.smallrye.config"/>
         <module name="io.smallrye.metrics"/>
+        <module name="io.smallrye.reactive.converters"/>
+        <module name="io.smallrye.common.annotation"/>
```
Comment (rhusar) [inline microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/fault-tolerance/main/module.xml:40]: Lets keep these alphabetical, so move this higher please.

File: testsuite/integration/microprofile-tck/fault-tolerance/pom.xml [hunk 3]
```diff
@@ -100,6 +100,7 @@
                         <!-- Increase timeout to reduce timing intermittent failures -->
                         <org.eclipse.microprofile.fault.tolerance.tck.timeout.multiplier>2.0</org.eclipse.microprofile.fault.tolerance.tck.timeout.multiplier>
                     </systemPropertyVariables>
+                    <runOrder>reversealphabetical</runOrder>
```
Comment (rhusar) [inline testsuite/integration/microprofile-tck/fault-tolerance/pom.xml:103]: Any particular reason for this?

**Review by rhusar — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (jasondlee) [reply pom.xml:316]: Done,

Comment (jasondlee) [reply pom.xml:316]: Updated.

File: microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/fault-tolerance/main/module.xml (hunk 2 above)
Comment (jasondlee) [reply microprofile/galleon-common/src/main/resources/modules/system/layers/base/io/smallrye/fault-tolerance/main/module.xml:40]: Fixed.

File: testsuite/integration/microprofile-tck/fault-tolerance/pom.xml (hunk 3 above)
Comment (jasondlee) [reply testsuite/integration/microprofile-tck/fault-tolerance/pom.xml:103]: MetricsDisabledTest and AllMetricsTest appear to be order-dependent, but not by design. It seems the same smallrye-metrics change that broke some tests for that spec upgrade is at work here. I made a change to our tests suggested by Ladislav Thon (https://github.com/smallrye/smallrye-fault-tolerance/blob/master/testsuite/tck/src/test/java/io/smallrye/faulttolerance/tck/CleanupMetricRegistries.java), but that failed to fix the issue. While this did, it's hardly ideal. I'm happy enough to stick with it for now if need be, but if you, being a bit more familiar with this part of the system, can find a better fix, I'd love to see it. :)

Comment (rhusar) [reply testsuite/integration/microprofile-tck/fault-tolerance/pom.xml:103]: Thanks, I will look into it.

**Comment by kabir** [comment]
@jasondlee I resolved a conflict via the GH editor. This results in another commit. Please check it and squash with the previous one.

@rhusar let me know when you guys agree this is ready

**Comment by jasondlee** [comment]
> @jasondlee I resolved a conflict via the GH editor. This results in another commit. Please check it and squash with the previous one.

I've pulled, squashed, pushed. :)

**Comment by rhusar** [comment]
@jasondlee Replaced by https://github.com/wildfly/wildfly/pull/13929 – that PR contains all the changes in here plus adds TCK and docs fixes. Thanks again Jason. 

@kabir FYI

**Comment by jasondlee** [comment]
That looks great. I'll close out this PR.

**Pull request closed without merging**